Improve load balancing

odbozhou opened the pull request:
1、Add rebalance thread to improve load balancing
2、Fix connector with same config duplicates without returning bugs

odbozhou committed:
Add rebalance thread

odbozhou committed:
Fix connector with same config duplicates without returning bugs

odbozhou committed:
Fix connector with same config duplicates without returning bugs

duhenglucky commented on openmessaging-runtime/src/main/java/io/openmessaging/connect/runtime/ConnectController.java:
```diff
@@ -132,32 +132,32 @@ public void start(){
                 log.error("schedule persist position error.", e);
             }
-        }, 1000, 20*1000, TimeUnit.MILLISECONDS);
+        }, 1000, 20 * 1000, TimeUnit.MILLISECONDS);
```
It would be nice if we can make this interval can be config.

duhenglucky commented on openmessaging-runtime/src/main/java/io/openmessaging/connect/runtime/service/RebalanceImpl.java:
```diff
@@ -76,44 +76,24 @@ public void doRebalance() {
         ConnAndTaskConfigs allocateResult = allocateConnAndTaskStrategy.allocate(curAliveWorkers.keySet(), worker.getWorkerId(), curConnectorConfigs, curTaskConfigs);
-        log.info("allocated connector:"+ allocateResult.getConnectorConfigs());
-        log.info("allocated task:"+ allocateResult.getTaskConfigs());
+        log.info("allocated connector:" + allocateResult.getConnectorConfigs());
```
We recommend the first letter in uppercase, and use placeholders({}) in the log instead "+".

duhenglucky commented on openmessaging-runtime/src/main/java/io/openmessaging/connect/runtime/service/RebalanceService.java:
```diff
@@ -0,0 +1,90 @@
+        log.info(this.getServiceName() + " service started");
+
+        while (!this.isStopped()) {
+            this.waitForRunning(20000);
```
Put this constant in the configuration file or environmental variable？

odbozhou committed:
1、Add two configurable variables（positionPersistInterval, configPersistInterval）.
2、Modify the non-standard log.
3、Rebalance interval time taking environment variables（odar.cluster.rebalance.waitInterval）.

odbozhou committed:
1、Modify the non-standard log.
2、Add a new environment variable（odar.max.message.size）to limit max message size.

odbozhou committed:
Limit the maximum message size

odbozhou committed:
Limit the maximum message size
